### PR TITLE
Update inputProvider to toggle on serviceName not language

### LINKED ISSUE #671: Update inputProvider to toggle on serviceName not language
**Is your feature request related to a problem? Please describe.**
Now that the font applied to a component will be based on the service #573 , we currently have to include `LANGUAGE_VARIANTS` to make a select knob in storybook. This means we will have two select input knobs with the same values and the user will have to select twice.

**Describe the solution you'd like**
Update `inputProvider` to have a dropdown of services, not languages, and then return the service name out in the provider for the components to use.

**Testing notes**
[Tester to complete]

Dev insight: Unit Tests will be sufficient

**Additional context**

Blocking https://github.com/bbc/psammead/issues/543

### PR DESCRIPTION
**Pull request by ghost**
Resolves #671

**Overall change:** Update inputProvider to toggle on serviceName not language.

**Code changes:**

- Update `inputProvider` to pass object `{ script, dir, service, inputs }`
- Add `service` to `dirDecorator`
- Update `inputProvider` tests

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: packages/utilities/psammead-storybook-helpers/src/index.stories.jsx [hunk 1]
```diff
@@ -22,7 +22,7 @@ storiesOf('Utilities|Input Provider', module)
         },
         { name: 'second slot' },
       ],
-      ([first, second], script, dir) => (
+      ({ inputs: [first, second], script, dir }) => (
```
Comment (dr3) [inline packages/utilities/psammead-storybook-helpers/src/index.stories.jsx]: Are these not outputs?

File: packages/utilities/psammead-storybook-helpers/README.md [hunk 2]
```diff
@@ -62,7 +63,7 @@ storiesOf('Caption', module)
         { name: 'caption', defaultText: 'Students sitting an examination' },
         { name: 'offscreen text', defaultText: 'Image Caption, ' },
       ],
-      ([captionText, offscreenText], script, dir) => (
+      ({ inputs: [captionText, offscreenText], script, dir }) => (
```
Comment (dr3) [inline packages/utilities/psammead-storybook-helpers/README.md]: Should this example have `service`?

**Review by dr3 — CHANGES_REQUESTED** [review]

File: packages/utilities/psammead-storybook-helpers/src/index.stories.jsx [hunk 3]
```diff
@@ -22,7 +22,7 @@ storiesOf('Utilities|Input Provider', module)
...
         <ul>
           <li>{first}</li>
           <li>{second}</li>
```
Comment (andrew-nowak) [inline packages/utilities/psammead-storybook-helpers/src/index.stories.jsx:28]: You could add an `<li>` showing off that you get the service name back from the input provider here.

There's also a few references in this file to 'language' or 'English' - they could be updated to 'service' and 'News'

File: packages/utilities/psammead-storybook-helpers/README.md (hunk 2 above)
Comment (ghost) [reply packages/utilities/psammead-storybook-helpers/README.md]: Yeah but we don't have service prop in the `Caption` component yet. But added it anyway 🦹‍♂️

File: packages/utilities/psammead-storybook-helpers/CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,7 @@
 <!-- prettier-ignore -->
 | Version | Description |
 |---------|-------------|
+| 3.0.0   | [PR#674](https://github.com/bbc/psammead/pull/674) Update inpuProvider to pass an object with script, service, dir and slotTexts |
```
Comment (DenisHdz) [inline packages/utilities/psammead-storybook-helpers/CHANGELOG.md]:
```suggestion
| 3.0.0   | [PR#674](https://github.com/bbc/psammead/pull/674) Update inputProvider to pass an object with script, service, dir and slotTexts |
```

**Review by DenisHdz — CHANGES_REQUESTED** [review]
Just a tiny typo

**Review by andrew-nowak — APPROVED** [review]
looks good

**Review by DenisHdz — APPROVED** [review]
LGTM 👍🏻

**Review by dr3 — APPROVED** [review]

**Comment by david-boydell** [comment]
Hi @sadickisaac, could you take a look at this please. When I checkout input-provider and view the StoryPromo > StoryPromo > Default I get the error `TypeError: arr[Symbol.iterator] is not a function`.

**Comment by david-boydell** [comment]
input-provider

**Comment by ghost** [comment]
Screenshot 2019-06-24 at 20 15 10

Can't reproduce the error. This is a breaking change and has not been pulled to any other packages. It should not affect other packages that use `inputProvider`. After checking out try `npm run ci:packages`

**Comment by david-boydell** [comment]
All good, unit tests all pass, this is ready for merge.

**Pull request merged**